### PR TITLE
docs: document controlled failure scenarios

### LINKED ISSUE #912: Follow-up to #607: docstrings for benchmark/controlled_failures.py
### Affected component

Benchmarks

### What problem does this solve?

`src/continuum/benchmark/controlled_failures.py` still has 2 undocumented public names (`ControlledScenario`, `by_name`). Same reason as #607: the diff-scoped check fails the next benchmark PR for code it did not write. `by_name` raises `KeyError` on unknown names, a contract currently stated nowhere.

How to reproduce:

```bash
python - <<EOF
import ast
from pathlib import Path
tree = ast.parse(Path("src/continuum/benchmark/controlled_failures.py").read_text(encoding="utf-8"))
print([n.name for n in ast.walk(tree)
       if isinstance(n, (ast.FunctionDef, ast.ClassDef))
       and not n.name.startswith("_")
       and ast.get_docstring(n) is None])
EOF
```

Expected: `['ControlledScenario', 'by_name']`.

Verified on current `main`: exactly those two. No open or closed issue or PR covers this file (#808 covers the different directory `benchmark/phase6` scenario builders).

### Proposed solution

Add docstrings in the #607 style: what a `ControlledScenario` row describes, and that `by_name` looks it up by scenario name raising `KeyError` when unknown. Docstrings only, no runtime change. Then run `ruff check`, `ruff format --check`, and `mypy src/continuum`.

### Alternatives considered

Returning `None` instead of raising `KeyError` so the docstring is unnecessary. Rejected: behavior change, out of scope for a docs issue.

### Is this a breaking change?

No

### Would you like to work on this?

Yes, with guidance

### PR DESCRIPTION
**Pull request by Adhi1-2**
## Summary
- document the `ControlledScenario` public data model
- document `by_name`, including its name-based lookup contract
- make the documentation-only benchmark API complete for future contributors

Closes #912

## Validation
- `./.venv/bin/pytest -q tests/test_controlled_failures.py` — 3 passed
- public-name AST audit reports no undocumented public classes or functions
- `./.venv/bin/ruff check src/continuum/benchmark/controlled_failures.py`
- `./.venv/bin/ruff format --check src/continuum/benchmark/controlled_failures.py`
- `./.venv/bin/mypy src/continuum/benchmark/controlled_failures.py`
- `git diff --check`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/continuum/benchmark/controlled_failures.py [hunk 1]
```diff
@@ -103,6 +105,8 @@ class ControlledScenario:
 
 
 def by_name(name: str) -> ControlledScenario:
+    """Look up a controlled failure scenario by name."""
```
Comment (Cyrax321) [inline src/continuum/benchmark/controlled_failures.py]: #912 explicitly requires the KeyError contract since nothing states it today, and the implementation raises `KeyError` on unknown names. Please state it so the lookup contract is complete.

```suggestion
    """Look up a controlled failure scenario by name, raising KeyError when unknown."""
```

**Review by Cyrax321 — CHANGES_REQUESTED** [review]
Hi @Adhi1-2, thank you for the contribution.

This documents the exact two names #912 lists, with no logic change. I verified the raise path in `by_name`, ran the test file with 3 passed, and confirmed ruff, format, and diff checks clean. CI is fully green.

Requesting changes for one small item inline: the `KeyError` contract #912 asks for. It is a one-line doc tweak with the wording ready. Once that is in, this is ready to approve.

**Comment by Adhi1-2** [comment]
Applied in fae966a — `by_name`'s docstring now reads `"""Look up a controlled failure scenario by name, raising KeyError when unknown."""`, exactly as suggested. ruff, format, and the controlled-failure tests remain green.

**Review by Cyrax321 — COMMENTED** [review]
Thank you for contributing to CONTINUUM. Two docstrings that pin the dataclass and its lookup exactly as implemented, and the jaccard edge cases spelled out. Appreciate the tight scope.

Verified: ControlledScenario described as one deterministic failure plus its expected recovery, and by_name raising KeyError when unknown, both matching src/continuum/benchmark/controlled_failures.py.

Not approving yet for process reasons only. Docs only plus 4 lines cannot fail Test (windows-latest, 3.11), so the branch is stale, and anya-review is red as well. Please rebase onto latest main, address the anya findings, and re-run. Happy to approve on green. Thanks for documenting the controlled failure surface.

**Comment by Adhi1-2** [comment]
Closing as a duplicate: issue #912 was closed by merged PR #1048 "docs(bench): docstrings for ControlledScenario and by_name" by @Adhi1-2, which lands the same change this branch carries.

Verified `gh issue view 912` is CLOSED and `gh pr view 1048` is MERGED, so this branch has no remaining value and is just queue noise. Deleting the branch to keep the fork tidy.
